python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -264,6 +268,7 @@ class GrapheneAssetNode(graphene.ObjectType):
         limit=graphene.Int(),
     )
     backfillPolicy = graphene.Field(GrapheneBackfillPolicy)
+    changedInBranchReasons = graphene.Field(non_null_list(GrapheneAssetChangedInBranchReason))
```

Comment:
i'm not sure about the "in branch" part since we may end up with other configurations of child parent deployment relationships that are not "branch" ie a long standing `staging` deployment child of `prod`

Reply:
fair. I thought `changedReasons` and `GrapheneAssetChangedReason` sounded too generic or that it was unclear that they were related to the changes between a parent and child deployment

Reply:
i may be over indexing on associating branch with branch deployments specifically 

I think its fine to use the terse names, but I could be convinced its better to use a more specific name

Reply:
i don't really have a strong preference either. Maybe once we lock down naming here https://github.com/dagster-io/dagster/pull/19681#discussion_r1491253628 that will make the naming for this class make more sense

Reply:
its pretty easy to change in the near term so I would say pick your favorite and run with it